pkg/apis/networking/validation/validation.go

```diff
@@ -561,6 +569,43 @@ func validateIngressTypedLocalObjectReference(params *api.TypedLocalObjectRefere
 	return allErrs
 }
 
+// validateIngressClassParametersReference ensures that Parameters fields are valid.
+// Parameters was previously of type `TypedLocalObjectReference` and used
+// `validateIngressTypedLocalObjectReference()`. This function extends validation
+// for additional fields introduced for namespace-scoped references.
+func validateIngressClassParametersReference(params *networking.IngressClassParametersReference, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if params == nil {
+		return allErrs
+	}
+
+	allErrs = append(allErrs, validateIngressTypedLocalObjectReference(&api.TypedLocalObjectReference{
+		APIGroup: params.APIGroup,
+		Kind:     params.Kind,
+		Name:     params.Name,
+	}, fldPath)...)
+
+	if params.Scope != "" || utilpointer.StringPtrDerefOr(params.Namespace, "") != "" ||
```

Comment:
@thockin:

> if gate-enabled && field not specified {
error: required
}

The current code is written with an assumption that this condition is not possible because when gate is enabled, the default kicks in.

In the current form, the if statement checks for two cases:
1. Gate is enabled, do validation
2. Field is specified (doesn't matter what is state of the gate), do validation

I think this simplifies the code a bit and provides the user enough validation as well.
Please also refer to earlier discussion in this PR with Rob https://github.com/kubernetes/kubernetes/pull/99275#discussion_r583341668 that provides some more background here.

Should we still change this or is it good as is?

Reply:
The validation code doesn't assume the defaulting code's semantics in general.  We should still validate the whole object.  For example, test cases might create objects directly that do not go through API defaulting and don't set this field and then crash later.

This has bitten me more than once.

Reply:
That makes sense to me. I'm working on updating the code.
Let's continue this in the new thread around the same block: https://github.com/kubernetes/kubernetes/pull/99275#discussion_r588516428